### PR TITLE
Update milli 0.29

### LINKED ISSUE #2144: Weird highlight results with non-ascii
**Describe the bug**
Meilisearch is giving weird results compared to algolia and typesense.

To be honest I don't know if the software is built to behave in this way, or the language I use is unsupported. I wanted to report this in any case. I there is a settings toggle for it to become strict, I'd like to know. I also uploaded the data to http://search.staging.academicpositions.com/ for you to try. (Not sure who owns the website though.)

**To Reproduce**
1. Go to meillisearch dashboard
2. Search for `"Müəllimlərin İşə Qəbulu - Təsviri İncəsənət"` or `"Təsviri İncəsənət"`
3. See the weird results

**Expected behavior**
Should show correct results as algolia and typesense.

**Screenshots**
Check out the video where I explain the results. Don't forget to enable audio.

https://user-images.githubusercontent.com/15037839/152649267-56b590cf-7093-457d-a167-339e4983fa61.mp4

if you want, here is youtube link https://www.youtube.com/watch?v=kumI6XbjnUA

**Meilisearch version:** v0.22.0, v0.25.2

### LINKED ISSUE #2375: Introduce the new version of the tokenizer: charabia
## Why?

We would make easier the contribution to our tokenizer in order to support more and more language. Indeed our community speaks multiple languages, we are not, and they are the best to choose which tokenizer and normalizer they want to use for their own native language. So @ManyTheFish worked on a new version of the tokenizer. More detailed in this issue https://github.com/meilisearch/charabia/issues/72

## What is fixed?

- Fixes this problem: https://github.com/meilisearch/product/discussions/151
- Make the tokenizer more contribuable

## Changes

- [x] Introduce the new version of the tokenizer
  - [x] Integrate it into milli
  - [x] Release milli
  - [x] Bump the new milli version into Meilisearch
- [x] Define precisely which languages we support or not, and how
- [x] Improve the issue management regarding the tokenizer (with @curquiza)

### LINKED ISSUE #2407: Improve ranking rules error message
**Describe the bug**
When changing the rules, the error message is misleading. According to the code :

https://github.com/meilisearch/meilisearch/blob/6d221058f19f9f262a99245882176ae9b7d48bec/meilisearch-http/tests/settings/get_settings.rs#L285-L290

Accepted rules are `Words, Typo, Sort, Proximity, Attribute, Exactness` in fact accepted rules are `words, typo, sort, proximity, attribute, exactness`

**To Reproduce**
Try sending a rule with a capital letter

**Expected behavior**
The rule name is case insensitive and the endpoint needs a fix and the error message is correct.

The rule name is case sensitive and the error message needs a fix to match this requirement.

**Meilisearch version:** [e.g. v0.27.1]

### LINKED ISSUE #2417: Hebrew language support
Just added by an external contributor: the support of the Hebrew language will be available in the new tokenizer version, [Charabia](https://github.com/meilisearch/charabia) 🎉 

Like for Chinese and Japanese, a normalizer and a segmenter have been added to [Charabia](https://github.com/meilisearch/charabia), making Meilisearch tokenizer Hebrew as expected (by the Hebrew community maintaining the normalizer and the segmenter).

If https://github.com/meilisearch/meilisearch/issues/2375 is closed, this issue is closed as well.

@meilisearch/docs-team

### PR DESCRIPTION
**Pull request by ManyTheFish**
- [x] Update milli to 0.29
- [x] Integrate charabia
- [x] Set disabled_words to default when Index::exact_words returns None
- [x] Fix ranking rules integration test

fixes #2375
fixes #2144
fixes #2417
fixes #2407

### CODE REVIEW

File: meilisearch-lib/src/index/index.rs [hunk 1]
```diff
@@ -175,12 +175,10 @@ impl Index {
...
-        let disabled_words = self
-            .exact_words(txn)?
-            .into_stream()
-            .into_strs()?
-            .into_iter()
-            .collect();
+        let disabled_words = match self.exact_words(txn)? {
+            Some(fst) => fst.into_stream().into_strs()?.into_iter().collect(),
+            None => Default::default(),
+        };
```
Comment (Kerollmops) [inline meilisearch-lib/src/index/index.rs:181]: We are sure that it is a `Vec` here, right? And that it doesn't cost anything to construct a default one. Can we change this line a little bit? Just to be sure, for the future.

```suggestion
        let disabled_words = match self.exact_words(txn)? {
            Some(fst) => fst.into_stream().into_strs()?.into_iter().collect(),
            None => Vec::new(),
        };
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Review by Kerollmops — APPROVED** [review]
I love this PR, it makes the code much easier to read, thank you!

bors merge

**Comment by ManyTheFish** [comment]
because windows wasn't ready for the `v0.29`, let's give it another chance.
bors merge

**Comment by ManyTheFish** [comment]
Three times in a row. 🤔
It begins to be annoying. I will rebase.

**Comment by ManyTheFish** [comment]
bors merge

**Comment by ManyTheFish** [comment]
Seriously? snapshots now? Ok round 9000:
bors merge

**Comment by curquiza** [comment]
bors merge

**Review by irevoire — DISMISSED** [review]

**Comment by irevoire** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by ManyTheFish** [comment]
last try before investigating:
bors merge

**Review by Kerollmops — DISMISSED** [review]

**Comment by ManyTheFish** [comment]
please. 🤞 

bors merge

**Review by Kerollmops — APPROVED** [review]
incro :chocolatine: yable!
bors merge

**Pull request merged**